test/hotspot/jtreg/compiler/vectorization/runner/ArrayTypeConvertTest.java

```diff
@@ -398,6 +426,10 @@ public short[] convertDoubleToShort() {
     @IR(applyIfCPUFeatureOr = {"sve", "true", "avx", "true"},
         applyIf = {"MaxVectorSize", ">=16"},
         counts = {IRNode.VECTOR_CAST_D2S, IRNode.VECTOR_SIZE + "min(max_double, max_char)", ">0"})
+    @IR(counts = {"castDtoX", " >0 "}, phase = CompilePhase.FINAL_CODE,
+        applyIfCPUFeatureAnd = {"avx", "true", "avx10_2", "false"})
+    @IR(counts = {"cast2DtoX", " >0 "}, phase = CompilePhase.FINAL_CODE,
+        applyIfCPUFeature = {"avx10_2", "true"})
```

Comment:
Please refer to https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/lib/ir_framework/IRNode.java#L2638 for adding MachNode IR node based checks

Reply:
Thanks, I added some new nodes.